spyder/widgets/projects/tests/test_projectdialog.py

```diff
@@ -28,5 +29,29 @@ def test_project_dialog(qtbot):
     assert dlg
 
 
+@patch('spyder.widgets.projects.projectdialog.getexistingdirectory')
+def test_projectdialog_location(mock_getexistingdirectory):
+    """Run getexistingdirectory with mock values"""
+    dlg = ProjectDialog(None)
+    mock_getexistingdirectory.return_value = r"c:\a/b\\c/d"
+    dlg.select_location()
+    assert dlg.location == r"c:\a\b\c\d"
```

Comment:
These tests will currently fail on non-Windows platforms (e.g. Travis CI), since they will be normalized to ``/``. You should either skip them (using ``@pytest.mark.skipif(os.name == 'nt'``), check for a different path depending on the platform, or just harmonize everything to ``/`` in your original function assuming others agree and then it'll work fine (another small reason for that approach).